app/services/catalog/next_name.rb

```diff
@@ -0,0 +1,17 @@
+module Catalog
+  class NextName
+    attr_reader :next_name
+
+    def initialize(portfolio_item_id, portfolio_id = nil)
+      @item = PortfolioItem.find(portfolio_item_id)
+      @portfolio = portfolio_id.present? ? Portfolio.find(portfolio_id) : @item.portfolio
+    end
+
+    def process
+      names = @portfolio.portfolio_items.pluck(:display_name)
+      @next_name = Catalog::NameAdjust.create_copy_name(@item.display_name, names)
+
+      self
```

Comment:
If the openapi spec just returns a string for next name, should we also just return  the name here instead of `self`?

Reply:
Service classes are suppose to return the service instance and not a specific value, so it should stay this way.

Reply:
Good point. Was looking at it from the perspective that our `openapi.json` is the source of truth so why send back the whole object.